docs/deployment/run-in-prod.md

```diff
@@ -0,0 +1,155 @@
+```
+The above command will start Gunicorn with 4 workers! 
+
+`uvicorn.workers.UvicornWorker` is the worker class we discussed about.
+The `UvicornWorker` implementation uses the `uvloop` and `httptools` implementations for improved performance. 
+
+To run under `PyPy` you'll want to use pure-python implementation instead. You can do this by using the `UvicornH11Worker` class.
+
+If you need to pass Uvicorn's config arguments to gunicorn workers then you'll have to subclass `UvicornWorker`.
+```python
+from uvicorn.workers import UvicornWorker
+
+class MyUvicornWorker(UvicornWorker):
+    CONFIG_KWARGS = {"loop": "asyncio", "http": "h11", "lifespan": "off"}
+```
+
+### Using Supervisor
+
+[Supervisor](http://supervisord.org/) is a client/server system that allows its users to monitor and control a number of processes on UNIX-like operating systems.
+
+To use `supervisor` as a process manager you should either:
+
+* Hand over the socket to uvicorn using its file descriptor, which supervisor always makes available as `0`, and which must be set in the `fcgi-program` section.
+* Or use a UNIX domain socket for each `uvicorn` process.
+
+A simple supervisor configuration might look something like this:
+
+**supervisord.conf**:
+
+```ini
+[supervisord]
+
+[fcgi-program:uvicorn]
+socket=tcp://localhost:8000
+command=venv/bin/uvicorn --fd 0 example:App
+numprocs=4
+process_name=uvicorn-%(process_num)d
+stdout_logfile=/dev/stdout
+stdout_logfile_maxbytes=0
+```
+
+Then run with `supervisord -n`.
+
+### Using Circus
+
+[Circus](https://circus.readthedocs.io/en/latest/) is a Python program which can be used to monitor and control processes and sockets.
+
+To use `circus` as a process manager, you should either:
+
+* Hand over the socket to uvicorn using its file descriptor, which circus makes available as `$(circus.sockets.web)`.
+* Or use a UNIX domain socket for each `uvicorn` process.
+
+A simple circus configuration might look something like this:
+
+**circus.ini**:
+
+```ini
+[watcher:web]
+cmd = venv/bin/uvicorn --fd $(circus.sockets.web) example:App
+use_sockets = True
+numprocesses = 4
+
+[socket:web]
+host = 0.0.0.0
+port = 8000
+```
+
+Then run `circusd circus.ini`.
+
+## Run using Docker Containers
+
+In this section we will see how to run your application using Docker.
+
+### Creating Docker Image
+
+Before getting started, make sure you have installed Docker in your environment. If not, follow the guide from [official documentation](https://docs.docker.com/get-started/#download-and-install-docker) to install.    
+
+While building a Docker image, the folder structure is very important. The following tutorial is using a structure like this:
+```
+.
+├── app
+│   ├── __init__.py
+│   └── main.py
+├── Dockerfile
+└── requirements.txt
+```
+
+Dockerfile:
+```commandline
+FROM python:3.9
+
+WORKDIR /code
+
+COPY ./requirements.txt /code/requirements.txt
+
+RUN pip install --no-cache-dir --upgrade -r /code/requirements.txt
```

Comment:
well, this is the kind of rabbit hole I was talking about when I begun reviewing this, let's teach poeple how to use docker...